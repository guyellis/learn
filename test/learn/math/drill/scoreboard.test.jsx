const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const db = require('../../../../src/learn/db');
const ScoreBoard = require('../../../../src/learn/math/drill/scoreboard');

const muiTheme = getMuiTheme(lightBaseTheme);

describe('ScoreBoard', () => {
  beforeEach(() => {
    db.getScores = jest.fn();
  });

  afterEach(() => {
    db.getScores.mockClear();
  });

  test('should render showScoreBoard', () => {
    db.getScores.mockReturnValueOnce([{
      levelIndex: 0,
      correctCount: 10,
      opIndexes: [0],
      timePerQuestion: 6,
    }]);

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <ScoreBoard />
      </MuiThemeProvider>,
    );

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });
});
