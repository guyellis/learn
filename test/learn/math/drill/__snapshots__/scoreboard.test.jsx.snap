// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard should render showScoreBoard 1`] = `
<div>
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
  >
    Scoreboard
  </h2>
  <div>
    <ul
      className="MuiList-root makeStyles-root-31 MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root"
        >
          <div
            aria-describedby={null}
            className="MuiAvatar-root MuiAvatar-colorDefault"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "backgroundColor": "#fbc02d",
              }
            }
            title="0 Gold"
          >
            0
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Gold Badge(s)
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            2 seconds or less (per question)
          </p>
        </div>
      </li>
      <li
        className="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
    </ul>
    <ul
      className="MuiList-root makeStyles-root-31 MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root"
        >
          <div
            aria-describedby={null}
            className="MuiAvatar-root MuiAvatar-colorDefault"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "backgroundColor": "#9e9e9e",
              }
            }
            title="0 Silver"
          >
            0
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Silver Badge(s)
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            between 2 and 3 seconds (per question)
          </p>
        </div>
      </li>
      <li
        className="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
    </ul>
    <ul
      className="MuiList-root makeStyles-root-31 MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root"
        >
          <div
            aria-describedby={null}
            className="MuiAvatar-root MuiAvatar-colorDefault"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "backgroundColor": "#8d6e63",
              }
            }
            title="0 Bronze"
          >
            0
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Bronze Badge(s)
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            between 3 and 4 seconds (per question)
          </p>
        </div>
      </li>
      <li
        className="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
    </ul>
    <ul
      className="MuiList-root makeStyles-root-31 MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root"
        >
          <div
            aria-describedby={null}
            className="MuiAvatar-root MuiAvatar-colorDefault"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "backgroundColor": "#81d4fa",
              }
            }
            title="1 Blue"
          >
            1
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Blue Badge(s)
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            more than 4 seconds (per question)
          </p>
        </div>
      </li>
      <li
        className="MuiDivider-root MuiDivider-inset"
        role="separator"
      />
    </ul>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <table
      className="MuiTable-root"
    >
      <thead
        className="MuiTableHead-root"
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Level
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Addition
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
      >
        <tr
          className="MuiTableRow-root"
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            scope="row"
          >
            A
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <span>
              <div
                aria-describedby={null}
                className="MuiAvatar-root MuiAvatar-colorDefault"
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "backgroundColor": "#81d4fa",
                  }
                }
                title="1 Blue"
              >
                1
              </div>
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
